pkgs/development/libraries/cxx-rs/default.nix

```diff
@@ -0,0 +1,55 @@
+{ cxx-rs, fetchFromGitHub, lib, rustPlatform, testers }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "cxx-rs";
+  version = "1.0.72";
+
+  src = fetchFromGitHub {
+    owner = "dtolnay";
+    repo = "cxx";
+    rev = version;
+    sha256 = "sha256-+OumeLSgz8kLQKhEc3icCk1q+X+S7Xt+XtAlkx8iguU=";
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+  };
```

Comment:
Can we avoid the lockfile by using fetchCrate?

Reply:
When using `fetchCrate` the tests fail [and the author has declined to fix them](https://github.com/dtolnay/cxx/issues/1071).

However, the tests do pass when the source is checked out from git. While I can patch `Cargo.toml` in this way:

```
[patch.crates-io]
cxx-test-suite = { git = "https://github.com/dtolnay/cxx.git" }
```

it leads to a conflict in the dependency tree:

```
├── cxx-test-suite v0.0.0 (https://github.com/dtolnay/cxx.git#9e0d3680)
│   └── cxx v1.0.72 (https://github.com/dtolnay/cxx.git#9e0d3680)
│       ├── cxxbridge-macro v1.0.72 (proc-macro) (https://github.com/dtolnay/cxx.git#9e0d3680)
```

`cxx-test-suite` is being compiled against the version of `cxx` which exists in git, not the version in the current working directory. I have not figured out a way to address this. Any ideas?